articles/introduction-to-reinforcement-learning/index.md

```diff
@@ -0,0 +1,109 @@
+
+RL is important especially when there is no sure way of performing a task. However, there are rules which the model has to follow to perform its duties correctly.
+
+### Key Concepts
+
+In RL we have key concepts that have to be defined. They include agents, state, action, rewards and the environment.
+
+1.	**Agent**. An agent is an autonomous entity which observes and acts upon an environment. The agent learns which actions to take by a process of trial and error. This process of trial and error requires the agent to explore his available actions on the fly. Whenever an agent tries an action, it will receive a reward which indicates how good or bad the selected action was. In the long run, an agent will try to maximize selecting actions that resulted in high rewards while attempting to reduce the actions which may lead to low rewards.
+
+2.	**State**. The state represents the decision-making factors under consideration being observed by the agent.
+
+3.	**Action**. An action represents an optimal action being selected by the agent, which may change or affect the state and reward.
+
+4.	**Reward**. A reward in RL represents the gains and losses in network performance for taking a particular action on a particular state in the previous time instance.
+
+5.	**Environment**. The environment is where the agent acts. As to how the environment reacts to certain actions is defined by a model which we may or may not know.
+
+![rl-architecture](/engineering-education/introduction-to-reinforcement-learning/rl-architecture.png)<br>
+*[Image Source: Springer Link](https://link.springer.com/chapter/10.1007/978-981-15-4095-0_2)*
+
+### Most used Reinforcement Learning Algorithms
+
+There are a couple of RL algorithms used in solving problems today. They include Q-learning, [Policy Gradients (PG)](https://papers.nips.cc/paper/1713-policy-gradient-methods-for-reinforcement-learning-with-function-approximation.pdf/) and [Actor-Critics](https://papers.nips.cc/paper/1786-actor-critic-algorithms.pdf/). Nonetheless, in this article, we tackle the popular Q-learning algorithm which has been used to solve many RL problems. 
+
+#### Q-learning (most popular).
+
+In the simplest form, Q-learning can be described as a value-based reinforcement learning algorithm used to find an optimal action-selection policy using a Q-function. The main goal is to maximize the value function Q by selecting the best of all possible actions.
+
+![q-learning](/engineering-education/introduction-to-reinforcement-learning/q-learning.PNG)<br>
+*[Image Source: Springer Link](https://link.springer.com/chapter/10.1007/978-981-15-4095-0_2)*
+
```

Comment:
There are terms in the above formula like 
'Learning Rate' and 'Discount Factor' that haven't been mentioned or explained. It might be a good idea to explain it briefly

Reply:
Fixed